utils/frame/remote-externalities/src/lib.rs

```diff
@@ -351,9 +351,53 @@ where
 		prefix: StorageKey,
 		at: B::Hash,
 	) -> Result<Vec<StorageKey>, &'static str> {
-		let mut last_key: Option<StorageKey> = None;
+		const NUM_PARTS: usize = 4; // number of parts to divide the prefix into
+
+		let mut tasks = Vec::with_capacity(NUM_PARTS);
+
+		// Divide the prefix into N parts
+		let prefix_bytes = prefix.0;
+		let total_len = prefix_bytes.len();
+		let part_len = total_len / NUM_PARTS;
```

Comment:
I dont get how this is supposed to work.  
Splitting a prefix happens by appending to it, not cutting it appart.

Reply:
I'm dividing the byte representation of the prefix into N parts by splitting it into equal parts of length `part_len`. This is not the same as appending to the prefix, as you correctly pointed out.

Can I split the prefix into four parts by changing the first byte of the key. Each part has a size of 256, which is the size of a byte. The start key of each range is set to the original prefix, and the end key is set by incrementing the first byte of the start key by 1.